shared/teams/team/rows/bot-row/bot/index.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react'
+import * as Kb from '../../../../../common-adapters'
+import * as Styles from '../../../../../styles'
+import {isLargeScreen} from '../../../../../constants/platform'
+import {MemberStatus, TeamRoleType} from '../../../../../constants/types/teams'
+import BotMenu from '../bot-menu'
+
+export type Props = {
+  botAlias: string
+  description: string
+  onClick: () => void
+  onEdit: () => void
+  onRemove: () => void
+  onShowTracker: () => void
+  ownerTeam?: string
+  ownerUser?: string
+  roleType: TeamRoleType
+  status: MemberStatus
+  username: string
+  youCanManageMembers: boolean
+}
+
+// NOTE the controls for reset and deleted users (and the chat button) are
+// duplicated here because the desktop & mobile layouts differ significantly. If
+// you're changing one remember to change the other.
+
+export const TeamBotRow = (props: Props) => {
+  let descriptionLabel
+  let menuRef = React.useRef<Kb.Box>(null)
+  const [showMenu, setShowMenu] = React.useState(false)
+
+  const _getAttachmentRef = () => menuRef.current
+  const _onShowMenu = () => setShowMenu(true)
+  const _onHideMenu = () => setShowMenu(false)
+  const active = props.status === 'active'
+  if (props.description.length > 0) {
+    descriptionLabel = (
+      <Kb.Text style={styles.fullNameLabel} type="BodySmall" lineClamp={1}>
+        {props.description}
+      </Kb.Text>
+    )
+  }
+
+  const usernameDisplay = (
+    <Kb.Box2 direction="horizontal" alignSelf="flex-start">
+      <Kb.Text
+        type="BodySmallSemibold"
+        style={{color: Styles.globalColors.black}}
+        onClick={props.onShowTracker}
+      >
+        {props.botAlias || props.username}
+      </Kb.Text>
+      <Kb.Text type="BodySmall">
+        &nbsp;• by{' '}
+        {props.ownerTeam ? (
+          <Kb.TeamWithPopup prefix="@" inline={true} teamName={props.ownerTeam} type="BodySmall" />
+        ) : (
+          <Kb.ConnectedUsernames
+            prefix="@"
+            inline={true}
+            usernames={[props.ownerUser ?? props.username]}
```

Comment:
Would "by _Unknown_" be better here?

Reply:
I think putting the username of the bot is nice so you can get the popup for the bot username (which might contain useful info about the author/source of the bot/whatever)